### DISCUSSION TITLE
Registration Not Working

### DISCUSSION THREAD

**Discussion by Theinventor17** [danny-avila/LibreChat#2315] [Help Wanted]
So i have an instance running using NPM Build but the registration doesn't work for certain domains; it only allows Gmail.com domains. Does anyone know how to fix this? I tried changing the allowed domains and registration and still getting the error.

**Comment by danny-avila** [comment]
what does your server startup logs look like? Post everything before `Server listening at http://host:port`

For example, mine:

```bash
danny@VENGEANCE:~/LibreChat$ npm run backend
> LibreChat@0.7.0 backend
> cross-env NODE_ENV=production node api/server/index.js
2024-04-04 18:56:40 info: [Optional] Redis not initialized. Note: Redis support is experimental.
2024-04-04 18:56:41 info: Connected to MongoDB
2024-04-04 18:56:41 info: Custom config file loaded:
2024-04-04 18:56:41 info: {
  "version": "1.0.5",
  "cache": true,
  "registration": {
    "socialLogins": [
      "github",
      "google",
      "discord"
    ],
    "allowedDomains": [
      "gmail.com",
      "outlook.com",
      "yahoo.com",
      "test.com"
    ]
  },
  "endpoints": {
    "azureOpenAI": {
# azure config
    },
    "custom": [
# custom endpoints
    ]
  }
}
2024-04-04 18:56:41 warn: RAG API is either not running or not reachable at http://localhost:8000, you may experience errors with file uploads.
Warning: connect.session() MemoryStore is not
designed for a production environment, as it will leak
memory, and will not scale past a single process.
2024-04-04 18:56:41 info: Server listening at http://localhost:3080
```

What does your custom config look like? Without `allowedDomains`, I can make up domains and they work. With `allowedDomains`, I am limited to the ones you see in my logs.

**Reply by Theinventor17** [reply]
this is what i have in the start up log 
![image](https://github.com/danny-avila/LibreChat/assets/111710441/1fbf1ea1-0262-4886-880b-781a3acde8bc)

**Reply by danny-avila** [reply]
```yaml
    "allowedDomains": [
      "gmail.com",
      "yahoo.com",
      "outlook.com",
      "test.xyz",
      "test.io",
      "test.co",
      "test.org",
      "test.net",
      "test.edu",
      "test.gov",
      "test.uk",
      "test.de",
      "test.gov",
      "test.info",
      "test.biz",
      "test.club",
      "test.online",
      "test.site",
      "test.store",
      "test.tech",
      "test.ai",
      "test.design",
      "test.me",
      "test.nl",
      "test.fr",
      "test.au"
    ]
  },
  ```
### This works
![image](https://github.com/danny-avila/LibreChat/assets/110412045/5560f587-253a-4280-bcfa-7cf54e1f4641)

### This doesn't, not on the list
![image](https://github.com/danny-avila/LibreChat/assets/110412045/0ec2a9d4-dadb-40b2-8c61-3c145ddfb67e)

**Reply by danny-avila** [reply]
are you on the latest update? how did you update? You run it with npm so curious what your update process looks like, that may be the issue

**Comment by Theinventor17** [comment]
Ok I got it to work the auth controller for allowed domains was set to false instead true. 

I do have another question, though: is there a way to change the order of the endpoints? So instead of it just being Open AI as the default endpoint

**Reply by ev-caseythefirst** [reply]
does the app only check allowed domains when using email registration? What about when social registration is enabled, and someone logs in with google? Does it check the allowed domains?

I have social registration on, with a restricted list of allowed domains, but I'm getting registrations from @gmail.com users, which is not in the allowedDomains list.

**Reply by danny-avila** [reply]
@ev-caseythefirst there was a merge for this on 4/24/24, are you at the latest commit?

https://github.com/danny-avila/LibreChat/pull/2512
